### PR TITLE
Distinguish between allVariables and allUsedVariables

### PR DESCRIPTION
**Pull request by rachel-fenichel**

### CODE REVIEW

**Comment by RoboErikG** [comment]
+1

File: core/variables.js [hunk 1]
```diff
@@ -75,6 +76,22 @@ Blockly.Variables.allVariables = function(root) {
 };
 
 /**
+ * Find all variables that the user has created through the workspace or
+ * toolbox.  For use by generators.
+ * @param {!Blockly.Workspace} root The workspace to inspect.
+ * @return {!Array.<string>} Array of variable names.
+ */
+Blockly.Variables.allVariables = function(root) {
+  if (root.getDescendants) {
```
Comment (NeilFraser) [inline core/variables.js]: Why not "root instanceof Blockly.Block"?  That way things don't get interesting if workspaces acquire a getDescendants function sometime in the future.

**Comment by NeilFraser** [comment]
LGTM other than the dangerous type check.

File: core/variables.js (hunk 1 above)
Comment (rachel-fenichel) [inline core/variables.js]: Because it's what was used before (old line 46/new line 47).  Want me to change both and then commit?

Comment (NeilFraser) [inline core/variables.js]: 46/47 is less about determining type, and more about verifying the existence of a function before calling it.  That could be done either way, but you are right, testing with instanceof would be a bit better.

**Pull request merged**